src/tools/ilasm/ILAssembler/GrammarVisitor.cs

```diff
@@ -0,0 +1,4351 @@
+                CILParser.INT8 => SignatureTypeCode.SByte,
+                CILParser.INT16 => SignatureTypeCode.Int16,
+                CILParser.INT32_ => SignatureTypeCode.Int32,
+                CILParser.INT64_ => SignatureTypeCode.Int64,
+                CILParser.UINT8 => SignatureTypeCode.Byte,
+                CILParser.UINT16 => SignatureTypeCode.UInt16,
+                CILParser.UINT32 => SignatureTypeCode.UInt32,
+                CILParser.UINT64 => SignatureTypeCode.UInt64,
+                _ => throw new UnreachableException()
+            });
+        }
+
+        GrammarResult ICILVisitor<GrammarResult>.VisitSlashedName(CILParser.SlashedNameContext context)
+        {
+            return VisitSlashedName(context);
+        }
+
+        public static GrammarResult.Literal<TypeName> VisitSlashedName(CILParser.SlashedNameContext context)
+        {
+            TypeName? currentTypeName = null;
+            foreach (var item in context.dottedName())
+            {
+                currentTypeName = new TypeName(currentTypeName, VisitDottedName(item).Value);
+            }
+            // We'll always have at least one dottedName, so the value here will be non-null
+            return new(currentTypeName!);
+        }
+
+        GrammarResult ICILVisitor<GrammarResult>.VisitSqstringSeq(CILParser.SqstringSeqContext context) => VisitSqstringSeq(context);
+
+        public static GrammarResult.FormattedBlob VisitSqstringSeq(CILParser.SqstringSeqContext context)
+        {
+            var strings = ImmutableArray.CreateBuilder<string?>(context.ChildCount);
+            foreach (var child in context.children)
+            {
+                string? str = null;
+
+                if (child is ITerminalNode { Symbol: { Type: CILParser.SQSTRING, Text: string stringValue } })
+                {
+                    str = stringValue;
+                }
+
+                strings.Add(str);
+            }
+            return new(strings.ToImmutable().SerializeSequence());
+        }
+
+        GrammarResult ICILVisitor<GrammarResult>.VisitStackreserve(CILParser.StackreserveContext context) => VisitStackreserve(context);
+        public GrammarResult.Literal<long> VisitStackreserve(CILParser.StackreserveContext context) => VisitInt64(context.int64());
+
+        GrammarResult ICILVisitor<GrammarResult>.VisitSubsystem(CILParser.SubsystemContext context) => VisitSubsystem(context);
+        public GrammarResult.Literal<int> VisitSubsystem(CILParser.SubsystemContext context) => VisitInt32(context.int32());
+
+        public GrammarResult VisitTerminal(ITerminalNode node) => throw new UnreachableException();
+        public GrammarResult VisitTls(CILParser.TlsContext context)
+        {
+            // TODO-SRM: System.Reflection.Metadata doesn't provide APIs to point a data declaration at a TLS slot or into the IL stream.
+            // We have tests for the TLS case (CoreCLR only supports it on Win-x86), but not for the IL case.
```

Comment:
Can we delete the support for the managed C++ TLS from ilasm (separate PR)? 

> into the IL stream

What does this mean?

Reply:
native ilasm has the ability to point a data declaration at 3 places:

1. read-only data.
2. TLS data.
3. The data stream with method bodies.

We have no tests for the 3rd case, but it does exist in the grammar. We can remove it (especially if we're removing the support for managed C++ TLS).

Reply:
In IL, it looks like the following:

```
.data cil Foo = int32(32)
```

I think below is the intended use case:

```
.method public static int32 main()
{
.data cil byearray ( 00 01 02 /* etc */ )
}
```

I think the intention was to allow you to specify arbitrary method bodies, but I don't think the emitting actually supports that correctly.